frontend/src/components/pages/details/components/DetailsReservationWidget/DetailsReservationWidget.tsx

```diff
@@ -13,6 +13,21 @@ interface DetailsReservationWidgetProps {
 
 declare let AllianceReseaux: any;
 
+const waitForGlobal = async (key: string) => {
+  return new Promise((resolve, reject) => {
+    // eslint-disable-next-line
+    if (window.hasOwnProperty(key)) {
+      resolve(true);
+    } else {
+      // eslint-disable-next-line
+      setTimeout(function () {
```

Comment:
For performance issues, maybe a `clearTimeout` is missing.

I wonder if it would be easier to use `Script` next component https://nextjs.org/docs/basic-features/script
and call the script through the `onLoad` attribute. Something like this:

```jsx
<Script
  src={`https://eitinerancecdn.open-system.fr/widgets-eiti/${project}/${project}.min.js`}
  strategy="lazyOnload"
  onLoad={onLoad}
/>
```

```jsx
const onLoad = useCallback(() => {
  (function (window, ITW) {
    if (ITW === undefined || window.AllianceReseaux === undefined) {
      return;
    }
    const layer = {
      map: { lib: 'MapLeafletDevice' },
      langue: language,
      partner: reservation.partner,
      routeId: id,
    };

    const spaClient = ITW.pages.getSinglePageApplicationClient({ layer });
    window.AllianceReseaux.jQuery(function () {
      spaClient.executePage();
    });
  })(window, window.eitinerance?.core);
}, [id, language, reservation]);
```